### PR TITLE
gnrc_ipv6_nib: make automatic NDP packet emission (NS/RS) configurable

### LINKED ISSUE #8476: tests: gnrc_netif fails on macOS
#### Description

The test application `tests/gnrc_netif` fails on macOS; but works on Linux.

#### Steps to reproduce the issue

run `make all test` in `tests/gnrc_netif`

#### Expected results

success, without errors like on Linux

```
/RIOT/tests/gnrc_netif/bin/native/tests_gnrc_netif.elf   
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

........................................................
OK (56 tests)
Sending data from Ethernet device:
00000000  3E  E6  B5  0F  19  23  3E  E6  B5  22  FD  0A  FF  FF  41  42
00000010  43  44  45  46  47  00
Sending data from Ethernet device:
00000000  FF  FF  FF  FF  FF  FF  3E  E6  B5  22  FD  0A  FF  FF  41  42
00000010  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  DC  00  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  22
00000010  19  0F  B5  E6  3E  31  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  D8  01  00  00  0B  FD  0A  FD  22  19  0F  B5  E6  3E  31
00000010  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  9C  02  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  31
00000010  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  9C  03  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  31
00000010  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  98  04  00  00  0B  FD  0A  FD  31  32  33  41  42  43  44
00000010  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  D8  05  00  00  FF  FF  0A  FD  22  19  0F  B5  E6  3E  31
00000010  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  98  06  00  00  FF  FF  0A  FD  31  32  33  41  42  43  44
00000010  45  46  47  00
Sending data from Ethernet device:
00000000  3E  E6  B5  0F  19  23  3E  E6  B5  22  FD  0A  86  DD  60  00
00000010  00  00  00  08  3B  40  FE  80  00  00  00  00  00  00  3C  E6
00000020  B5  FF  FE  22  FD  0A  FE  80  00  00  00  00  00  00  3C  E6
00000030  B5  FF  FE  0F  19  23  41  42  43  44  45  46  47  00
Sending data from Ethernet device:
00000000  33  33  00  00  00  01  3E  E6  B5  22  FD  0A  86  DD  60  00
00000010  00  00  00  08  3B  40  FE  80  00  00  00  00  00  00  3C  E6
00000020  B5  FF  FE  22  FD  0A  FF  02  00  00  00  00  00  00  00  00
00000030  00  00  00  00  00  01  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  DC  07  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  22
00000010  19  0F  B5  E6  3E  60  00  00  00  00  08  3B  40  FE  80  00
00000020  00  00  00  00  00  3C  E6  B5  0F  19  22  FD  0A  FE  80  00
00000030  00  00  00  00  00  3C  E6  B5  FF  FE  0F  19  23  41  42  43
00000040  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  D8  08  00  00  FF  FF  0A  FD  22  19  0F  B5  E6  3E  60
00000010  00  00  00  00  08  3B  40  FE  80  00  00  00  00  00  00  3C
00000020  E6  B5  0F  19  22  FD  0A  FF  02  00  00  00  00  00  00  00
00000030  00  00  00  00  00  00  01  41  42  43  44  45  46  47  00
pktdump dumping Ethernet packet with empty payload
PKTDUMP: data received:
~~ SNIP  0 - size:   0 byte, type: NETTYPE_UNDEF (0)
00000000~~ SNIP  1 - size:  22 byte, type: NETTYPE_NETIF (-1)
if_pid: 6  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: 3e:e6:b5:22:fd:0b
dst_l2addr: 3e:e6:b5:22:fd:0a
~~ PKT    -  2 snips, total size:  22 byte
pktdump dumping IEEE 802.15.4 packet with empty payload
PKTDUMP: data received:
~~ SNIP  0 - size:   0 byte, type: NETTYPE_UNDEF (0)
00000000~~ SNIP  1 - size:  26 byte, type: NETTYPE_NETIF (-1)
if_pid: 7  rssi: -12744  lqi: 7
flags: 0x0
src_l2addr: 3e:e6:b5:0f:19:22:fd:0b
dst_l2addr: 3e:e6:b5:0f:19:22:fd:0a
~~ PKT    -  2 snips, total size:  26 byte
pktdump dumping Ethernet packet with payload 12 34 45 56
PKTDUMP: data received:
~~ SNIP  0 - size:   4 byte, type: NETTYPE_UNDEF (0)
00000000  12  34  45  56
~~ SNIP  1 - size:  22 byte, type: NETTYPE_NETIF (-1)
if_pid: 6  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: 3e:e6:b5:22:fd:0b
dst_l2addr: 3e:e6:b5:22:fd:0a
~~ PKT    -  2 snips, total size:  26 byte
pktdump dumping IEEE 802.15.4 packet with payload 12 34 45 56
PKTDUMP: data received:
~~ SNIP  0 - size:   4 byte, type: NETTYPE_UNDEF (0)
00000000  12  34  45  56
~~ SNIP  1 - size:  26 byte, type: NETTYPE_NETIF (-1)
if_pid: 7  rssi: -12744  lqi: 7
flags: 0x0
src_l2addr: 3e:e6:b5:0f:19:22:fd:0b
dst_l2addr: 3e:e6:b5:0f:19:22:fd:0a
~~ PKT    -  2 snips, total size:  30 byte
pktdump dumping IPv6 over Ethernet packet with payload 01
PKTDUMP: data received:
~~ SNIP  0 - size:  41 byte, type: NETTYPE_IPV6 (2)
traffic class: 0x00 (ECN: 0x0, DSCP: 0x00)
flow label: 0x00000
length: 1  next header: 59  hop limit: 64
source address: fe80::3fe6:b5ff:fe22:fd0a
destination address: fe80::3fe6:b5ff:fe22:fd0b
~~ SNIP  1 - size:  22 byte, type: NETTYPE_NETIF (-1)
if_pid: 6  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: 3e:e6:b5:22:fd:0b
dst_l2addr: 3e:e6:b5:22:fd:0a
~~ PKT    -  2 snips, total size:  63 byte
PKTDUMP: data to send:
~~ SNIP  0 - size:  10 byte, type: NETTYPE_NETIF (-1)
if_pid: 9  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: (nil)
dst_l2addr: (nil)
~~ SNIP  1 - size:  40 byte, type: NETTYPE_IPV6 (2)
traffic class: 0x00 (ECN: 0x0, DSCP: 0x00)
flow label: 0x00000
length: 53271  next header: 255  hop limit: 255
source address: ::
destination address: ff02::2
~~ SNIP  2 - size:   8 byte, type: NETTYPE_ICMPV6 (3)
   type: 133  code:   0
   cksum: 0x7bb8
~~ PKT    -  3 snips, total size:  58 byte
Sending data from Ethernet device:
00000000  33  33  00  00  00  02  3E  E6  B5  22  FD  0A  86  DD  60  00
00000010  00  00  00  08  3A  FF  00  00  00  00  00  00  00  00  00  00
00000020  00  00  00  00  00  00  FF  02  00  00  00  00  00  00  00  00
00000030  00  00  00  00  00  02  85  00  7B  B8  00  00  00  00
PKTDUMP: data to send:
~~ SNIP  0 - size:  10 byte, type: NETTYPE_NETIF (-1)
if_pid: 6  rssi: 0  lqi: 0
flags: 0
```

#### Actual results

fails with on MacOS:

```
/RIOT/tests/gnrc_netif/bin/native/tests_gnrc_netif.elf   
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

........................................................
OK (56 tests)
Sending data from Ethernet device:
00000000  3E  E6  B5  0F  19  23  3E  E6  B5  22  FD  0A  FF  FF  41  42
00000010  43  44  45  46  47  00
Sending data from Ethernet device:
00000000  FF  FF  FF  FF  FF  FF  3E  E6  B5  22  FD  0A  FF  FF  41  42
00000010  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  DC  00  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  22
00000010  19  0F  B5  E6  3E  31  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  D8  01  00  00  0B  FD  0A  FD  22  19  0F  B5  E6  3E  31
00000010  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  9C  02  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  31
00000010  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  98  03  00  00  FF  FF  0A  FD  7B  4B  3A  02  85  00  7B
00000010  B8  00  00  00  00
Sending data from Ethernet device:
00000000  33  33  00  00  00  02  3E  E6  B5  22  FD  0A  86  DD  60  00
00000010  00  00  00  08  3A  FF  00  00  00  00  00  00  00  00  00  00
00000020  00  00  00  00  00  00  FF  02  00  00  00  00  00  00  00  00
00000030  00  00  00  00  00  02  85  00  7B  B8  00  00  00  00
PKTDUMP: data to send:
~~ SNIP  0 - size:  10 byte, type: NETTYPE_NETIF (-1)
if_pid: 8  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: (nil)
dst_l2addr: (nil)
~~ SNIP  1 - size:  40 byte, type: NETTYPE_IPV6 (2)
traffic class: 0x00 (ECN: 0x0, DSCP: 0x00)
flow label: 0x00000
length: 53271  next header: 255  hop limit: 255
source address: ::
destination address: ff02::2
~~ SNIP  2 - size:   8 byte, type: NETTYPE_ICMPV6 (3)
   type: 133  code:   0
   cksum: 0x7bb8
~~ PKT    -  3 snips, total size:  58 byte
PKTDUMP: data to send:
~~ SNIP  0 - size:  10 byte, type: NETTYPE_NETIF (-1)
if_pid: 7  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: (nil)
dst_l2addr: (nil)
~~ SNIP  1 - size:  40 byte, type: NETTYPE_IPV6 (2)
traffic class: 0x00 (ECN: 0x0, DSCP: 0x00)
flow label: 0x00000
length: 22551  next header: 255  hop limit: 255
source address: ::
destination address: ff02::2
~~ SNIP  2 - size:   8 byte, type: NETTYPE_ICMPV6 (3)
   type: 133  code:   0
   cksum: 0x7bb8
~~ PKT    -  3 snips, total size:  58 byte
PKTDUMP: data to send:
~~ SNIP  0 - size:  10 byte, type: NETTYPE_NETIF (-1)
if_pid: 6  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: (nil)
dst_l2addr: (nil)
~~ SNIP  1 - size:  40 byte, type: NETTYPE_IPV6 (2)
traffic class: 0x00 (ECN: 0x0, DSCP: 0x00)
flow label: 0x00000
length: 57366  next header: 255  hop limit: 255
source address: ::
destination address: ff02::2
~~ SNIP  2 - size:   8 byte, type: NETTYPE_ICMPV6 (3)
   type: 133  code:   0
   cksum: 0x7bb8
~~ PKT    -  3 snips, total size:  58 byte
PKTDUMP: data to send:
~~ SNIP  0 - size:  10 byte, type: NETTYPE_NETIF (-1)
if_pid: 9  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: (nil)
dst_l2addr: (nil)
~~ SNIP  1 - size:  40 byte, type: NETTYPE_IPV6 (2)
traffic class: 0x00 (ECN: 0x0, DSCP: 0x00)
flow label: 0x00000
length: 26646  next header: 255  hop limit: 255
source address: ::
destination address: ff02::2
~~ SNIP  2 - size:   8 byte, type: NETTYPE_ICMPV6 (3)
   type: 133  code:   0
   cksum: 0x7bb8
~~ PKT    -  3 snips, total size:  58 byte
Sending data from IEEE 802.15.4 device:
00000000  41  9C  04  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  31
00000010  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  98  05  00  00  0B  FD  0A  FD  31  32  33  41  42  43  44
00000010  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  D8  06  00  00  FF  FF  0A  FD  22  19  0F  B5  E6  3E  31
00000010  32  33  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  98  07  00  00  FF  FF  0A  FD  31  32  33  41  42  43  44
00000010  45  46  47  00
Sending data from Ethernet device:
00000000  3E  E6  B5  0F  19  23  3E  E6  B5  22  FD  0A  86  DD  60  00
00000010  00  00  00  08  3B  40  FE  80  00  00  00  00  00  00  3C  E6
00000020  B5  FF  FE  22  FD  0A  FE  80  00  00  00  00  00  00  3C  E6
00000030  B5  FF  FE  0F  19  23  41  42  43  44  45  46  47  00
Sending data from Ethernet device:
00000000  33  33  00  00  00  01  3E  E6  B5  22  FD  0A  86  DD  60  00
00000010  00  00  00  08  3B  40  FE  80  00  00  00  00  00  00  3C  E6
00000020  B5  FF  FE  22  FD  0A  FF  02  00  00  00  00  00  00  00  00
00000030  00  00  00  00  00  01  41  42  43  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  DC  08  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  22
00000010  19  0F  B5  E6  3E  60  00  00  00  00  08  3B  40  FE  80  00
00000020  00  00  00  00  00  3C  E6  B5  0F  19  22  FD  0A  FE  80  00
00000030  00  00  00  00  00  3C  E6  B5  FF  FE  0F  19  23  41  42  43
00000040  44  45  46  47  00
Sending data from IEEE 802.15.4 device:
00000000  41  D8  09  00  00  FF  FF  0A  FD  22  19  0F  B5  E6  3E  60
00000010  00  00  00  00  08  3B  40  FE  80  00  00  00  00  00  00  3C
00000020  E6  B5  0F  19  22  FD  0A  FF  02  00  00  00  00  00  00  00
00000030  00  00  00  00  00  00  01  41  42  43  44  45  46  47  00
pktdump dumping Ethernet packet with empty payload
PKTDUMP: data received:
~~ SNIP  0 - size:   0 byte, type: NETTYPE_UNDEF (0)
00000000~~ SNIP  1 - size:  22 byte, type: NETTYPE_NETIF (-1)
if_pid: 6  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: 3e:e6:b5:22:fd:0b
dst_l2addr: 3e:e6:b5:22:fd:0a
~~ PKT    -  2 snips, total size:  22 byte
pktdump dumping IEEE 802.15.4 packet with empty payload
PKTDUMP: data received:
~~ SNIP  0 - size:   0 byte, type: NETTYPE_UNDEF (0)
00000000~~ SNIP  1 - size:  26 byte, type: NETTYPE_NETIF (-1)
if_pid: 7  rssi: -2360  lqi: 16
flags: 0x0
src_l2addr: 3e:e6:b5:0f:19:22:fd:0b
dst_l2addr: 3e:e6:b5:0f:19:22:fd:0a
~~ PKT    -  2 snips, total size:  26 byte
pktdump dumping Ethernet packet with payload 12 34 45 56
PKTDUMP: data received:
~~ SNIP  0 - size:   4 byte, type: NETTYPE_UNDEF (0)
00000000  12  34  45  56
~~ SNIP  1 - size:  22 byte, type: NETTYPE_NETIF (-1)
if_pid: 6  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: 3e:e6:b5:22:fd:0b
dst_l2addr: 3e:e6:b5:22:fd:0a
~~ PKT    -  2 snips, total size:  26 byte
pktdump dumping IEEE 802.15.4 packet with payload 12 34 45 56
PKTDUMP: data received:
~~ SNIP  0 - size:   4 byte, type: NETTYPE_UNDEF (0)
00000000  12  34  45  56
~~ SNIP  1 - size:  26 byte, type: NETTYPE_NETIF (-1)
if_pid: 7  rssi: -2360  lqi: 16
flags: 0x0
src_l2addr: 3e:e6:b5:0f:19:22:fd:0b
dst_l2addr: 3e:e6:b5:0f:19:22:fd:0a
~~ PKT    -  2 snips, total size:  30 byte
pktdump dumping IPv6 over Ethernet packet with payload 01
PKTDUMP: data received:
~~ SNIP  0 - size:  41 byte, type: NETTYPE_IPV6 (2)
traffic class: 0x00 (ECN: 0x0, DSCP: 0x00)
flow label: 0x00000
length: 1  next header: 59  hop limit: 64
source address: fe80::3fe6:b5ff:fe22:fd0a
destination address: fe80::3fe6:b5ff:fe22:fd0b
~~ SNIP  1 - size:  22 byte, type: NETTYPE_NETIF (-1)
if_pid: 6  rssi: 0  lqi: 0
flags: 0x0
src_l2addr: 3e:e6:b5:22:fd:0b
dst_l2addr: 3e:e6:b5:22:fd:0a
~~ PKT    -  2 snips, total size:  63 byte
Timeout in expect script at "child.expect("00000000  41  9C  03  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  31")" (tests/gnrc_netif/tests/01-run.py:40)
  File "/Volumes/devel/github/smlng/RIOT/dist/tools/testrunner/testrunner.py", line 55, in run
    testfunc(child)
  File "tests/01-run.py", line 40, in testfunc
    child.expect("00000000  41  9C  03  00  00  0B  FD  22  19  0F  B5  E6  3E  0A  FD  31")
  File "/usr/local/lib/python3.6/site-packages/pexpect/spawnbase.py", line 321, in expect
    timeout, searchwindowsize, async)
  File "/usr/local/lib/python3.6/site-packages/pexpect/spawnbase.py", line 345, in expect_list
    return exp.expect_loop(timeout)
  File "/usr/local/lib/python3.6/site-packages/pexpect/expect.py", line 107, in expect_loop
    return self.timeout(e)
  File "/usr/local/lib/python3.6/site-packages/pexpect/expect.py", line 70, in timeout
    raise TIMEOUT(msg)

make: *** [test] Error 1
```

#### Versions

OS= macOS 10.13.3 with:

```
Installed toolchain versions
----------------------------
          native gcc: gcc (Homebrew GCC 7.2.0_1) 7.2.0
          msp430-gcc: missing/error
             avr-gcc: missing/error
   arm-none-eabi-gcc: arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2017-q4-major) 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
    mips-mti-elf-gcc: missing/error
               clang: Apple LLVM version 9.0.0 (clang-900.0.39.2)
arm-none-eabi-newlib: "2.5.0"
 mips-mti-elf-newlib: missing/error
            avr-libc: missing/error (missing/error)
            cppcheck: Cppcheck 1.82
          coccinelle: spatch version 1.0.6 compiled with OCaml version 4.05.0
                 git: git version 2.16.1
```

Note: even though GCC 7.2.0 is installed, compiler for native is clang.

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
The `gnrc_netif` test doesn't test NDP behavior so emissions of NDP neighbor and router solicitations by the NIB is more harmful than helpful. Because of that I allowed for disabling RS and also fixed a bug in the NIB's SLAAC submodule where disabling wasn't possible, since `addr_str` was defined outside of an `ifdef`.

Please also test locally, since for some `samr21-xpro` the test works already, due to the NDP packets being send randomized (with a seed based on the CPU ID).

### Issues/PRs references
Fixes #8476 and https://github.com/RIOT-OS/RIOT/pull/9007#issuecomment-387421271, required for #9669.

### CODE REVIEW

**Comment by miri64** [comment]
(`saclay,samr21,2` is known to be failing on this test in master, while `saclay,samr21,1` is working e.g.)

**Comment by cladmi** [comment]
Changes looks sensible, I am going to test with several `iotlab-m3` and `samr21-xpro` on IoT-LAB.

File: tests/gnrc_netif/Makefile [hunk 1]
```diff
@@ -21,6 +21,10 @@ USEMODULE += netdev_ieee802154
 USEMODULE += netdev_test
 USEMODULE += od
 
+# deactivate automatically emitted packets from IPv6 neighbor discovery
+CFLAGS += -DGNRC_IPV6_NIB_CONF_ARSM=1
```
Comment (smlng) [inline tests/gnrc_netif/Makefile]: looks unrelated, at least not used above

Comment (miri64) [reply tests/gnrc_netif/Makefile]: the address state machine causes sending of unsolicited neighbor solicitations.

Comment (miri64) [reply tests/gnrc_netif/Makefile]: Arghs, but it shoud be 0. You are right!

**Comment by miri64** [comment]
Even though the tests don't trigger the address resolution state machine at the moment, it still makes sense to deactivate it, in case later added tests trigger it.

**Comment by miri64** [comment]
(The ARSM also causes the NIB to emit packets, so that is why it belongs in the *disable NDP emissions* commit ;-))

**Comment by cladmi** [comment]
Murdock failed on an unrelated error.

I think you can also squash.

**Comment by miri64** [comment]
Squashed.

**Comment by miri64** [comment]
@smlng @cladmi I leave merging to either of you.

**Comment by miri64** [comment]
Backport provided in #9732
